plugins/modules/ec2_instance.py

```diff
@@ -1736,19 +1762,20 @@ def build_top_level_options(params):
     return spec
 
 
-def build_instance_tags(params):
+def build_instance_tags(params: Dict[str, Any]) -> Optional[Dict[str, Any]]:
```

Comment:
Since boto3_tag_specifications returns either a list of dicts or an empty list https://github.com/ansible-collections/amazon.aws/blob/main/plugins/module_utils/tagging.py#L162, shouldn't the typing of the returned values be a list and not optional?